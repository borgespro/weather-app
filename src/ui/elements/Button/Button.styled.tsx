import styled from 'styled-components/native';

import { CustomThemeProps } from '@/typings/theme';

type InnerComponentsProps = {
  active: boolean;
};

export const ButtonComponent = styled.TouchableOpacity<CustomThemeProps & InnerComponentsProps>`
  min-width: ${(props) => props.theme.spacing.xxxl};
  border-width: 1px;
  border-radius: ${(props) => props.theme.spacing.m};
  border-color: ${(props) => props.theme.colors.secondary};
  background-color: ${(props) => (props.active ? props.theme.colors.secondary : props.theme.colors.background)};
`;

export const ButtonLabel = styled.Text<CustomThemeProps & InnerComponentsProps>`
  font-weight: bold;
  color: ${(props) => (props.active ? props.theme.colors.primary : props.theme.colors.secondary)};
  padding-vertical: ${(props) => props.theme.spacing.xxs};
  padding-horizontal: ${(props) => props.theme.spacing.l};
`;
