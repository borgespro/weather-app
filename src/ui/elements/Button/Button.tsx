import React from 'react';
import { GestureResponderEvent } from 'react-native';

import { ButtonComponent, ButtonLabel } from './Button.styled';

type ButtonProps = {
  label: string;
  onPress: (event?: GestureResponderEvent) => void;
  active?: boolean;
};

export default function Button({ label, active, onPress }: ButtonProps): React.ReactElement {
  return (
    <ButtonComponent active={!!active} onPress={onPress}>
      <ButtonLabel active={!!active}>{label}</ButtonLabel>
    </ButtonComponent>
  );
}
