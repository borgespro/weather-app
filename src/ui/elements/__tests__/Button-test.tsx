import React from 'react';

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Button from '../Button';

describe('Button Element', () => {
  it('should render without issues', () => {
    const onPress = jest.fn();
    const component = shallow(<Button label={'Test'} onPress={onPress} />);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
    const label = component.children().first().text();
    expect(label).toEqual('Test');
  });

  it('should be call onPress events', () => {
    const onPress = jest.fn();
    const wrapper = shallow(<Button label={'Test'} onPress={onPress} />);
    wrapper.props().onPress();
    expect(onPress).toHaveBeenCalled();
  });
});
