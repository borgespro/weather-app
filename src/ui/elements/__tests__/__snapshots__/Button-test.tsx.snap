// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Element should render without issues 1`] = `
<Styled(Component)
  active={false}
  onPress={[MockFunction]}
>
  <Styled(Text)
    active={false}
  >
    Test
  </Styled(Text)>
</Styled(Component)>
`;
